src/components/recordset/table-row.tsx

```diff
@@ -153,30 +156,59 @@ const TableRow = ({
     if (sensor && rowValues.length === 1 && justOverflows.length === 0) sensor.detach();
   }
 
-  // TODO: This assumes that tuple is set before rowValues. And that useEffect triggers before useLayoutEffect
+  // This assumes that tuple is set before rowValues. And that useEffect triggers before useLayoutEffect
   // NOTE: if the tuple changes, the table-row component isn't destroyed so the overflows need to be reset
   useEffect(() => {
     if (disableMaxRowHeightFeature) return;
     setOverflow([]);
   }, [tuple]);
 
+  // attach resize sensor to the table row and an onload event to each <img> tag in the table row
+  // onload will update a state variable to communicate when all images have loaded to trigger overflow logic once more
   useLayoutEffect(() => {
     if (!rowContainer.current || disableMaxRowHeightFeature) return;
     const tempSensor = new ResizeSensor(
       rowContainer.current,
-      () => setTimeout(() => {
-        initializeOverflows()
-        // images take ~200ms to load so wait at least that long before running
-      }, 200)
+      () => initializeOverflows()
     )
 
     setSensor(tempSensor);
+    
+    // fetch all <img> tags with -chaise-post-load class and keep count of the total
+    // attach an onload function that updates how many have loaded
+    const imgTags = rowContainer.current.querySelectorAll('img.-chaise-post-load');
+    setNumImages(imgTags.length);
+    
+    imgTags.forEach((image: HTMLImageElement) => {
+      image.onload = () => {
```

Comment:
Let's use `addEventListener` and `removeEventListener` instead of `onload`.

Reply:
I don't think you need to have `numImages` and `numImagesLoaded` state variables. They can just be `useRef`, and then in the `load` event listener you would check whether all the images have been loaded and run the `initializeOverflows` function.